src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -276,9 +284,25 @@ private static string GenerateUniqueChannelLocalName()
 
                 lock (_sessionGuard)
```

Comment:
Yes, and that's the problem. What is the lock actually guarding. Certainly not session.